app/src/org/commcare/network/RemoteDataPullResponse.java

```diff
@@ -111,31 +112,14 @@ public void notifyCurrentCount(long bytesRead) {
     }
 
     protected InputStream getInputStream() throws IOException {
-        return AndroidHttpClient.getUngzippedContent(response.getEntity());
-    }
-
-    public String getShortBody() throws IOException {
-        return new String(StreamsUtil.inputStreamToByteArray(AndroidHttpClient.getUngzippedContent(response.getEntity())));
+        return response.body().byteStream();
```

Comment:
So I had confirmed that the request does contain the header for gzip encoding which basically means that there should not be any change in response. I am not sure though how can I test it with the returned HTTP response. Should I check for a size decrease or something like that!

Reply:
Checking the response for the `Content-Encoding: gzip` header should be enough, i'd think, but we shouldn't block on it if you already validated that we're sending up the gzip request, since there's no reason to believe the server would respond differently. (We can double check next week to ensure we're getting a gzip'd response content encoding)

Reply:
Verified `Content-Encoding: gzip` in response as well.